rasterio/_features.pyx

```diff
@@ -62,7 +62,7 @@ def _shapes(image, mask, connectivity, transform):
     is_float = _getnpdtype(image.dtype).kind == "f"
     fieldtp = 2 if is_float else 0
 
-    valid_dtypes = ('int16', 'int32', 'uint8', 'uint16', 'float32')
+    valid_dtypes = ("int8", "int16", "int32", "uint8", "uint16", "float32", "float64")
```

Comment:
@theroggy I'd like to see int8 and float64 added to this list only for the appropriate GDAL versions. As in https://github.com/rasterio/rasterio/blob/main/rasterio/features.py#L264-L267. I'm not sure if the versions are the same for rasterizing and polygonizing. GDAL's change log isn't clear about this, it would take a little research in the GDAL issue tracker.

Reply:
@sgillies added conditional for `int8`. `float64` seems to be working for all versions of GDAL that the CI tests are running for (so >= 3.3).